design/user_data_incentives.md

```diff
@@ -0,0 +1,124 @@
+
+They can perform this service relatively cheap, because all data is visible to everyone, so they can employ efficient caching techniques. 
+They also personally gain from understanding the user queries, and from analysing the exposed data.
+
+As a result, the crypto users have become used to this as a free service, and take it for granted.
+
+The data that is queried by users is:
+
+1. Transaction receipts. Either queried after submitting a transaction to understand whether it was executed successfully and included in a block, 
+   or to retrieve the submitted transactions.
+2. RPC "eth_call" - which are used to query the status of the various smart contracts. For example, by far the most used: "ERC20.getBalance"   
+3. RPC "get_balance" - to query the balance of the native ETH for an account.
+4. Event query or subscription - used by dApps to update the UIs 
+
+
+## Chains with privacy
+
+It gets a bit more complicated when privacy is added to the mix, because the results of the user interactions are dependent on the requester.
+Only logic run inside an approved TEE is able to return a result, since it is the only place where that data is visible.
+
+This means that it is now much more difficult to create a cheap caching layer.
+Also, the service provider will not gain any insights from this service since all traffic and processing is encrypted.
+
+
+## Delegate to the nodes
+
+The usual rescue in the crypto space is to employ incentives.
+
+Obscuro is a decentralised network of nodes with different roles who already have their own incentives (Aggregators and Verifiers).
+
+The ideal scenario is to have a large and diverse community of verifiers node to make sure that the network functions correctly.
+
+Is is natural to assign to the verifier nodes the additional task of servicing user requests.
+
+### Incentives
+
+There has to be a reason for a node operator to provide this service to users. If there is no payoff, then most operators could just block the ports
+and not accept incoming connections.
+
+Here is a list and analyisis of various potential reasons for offering this service
+
+#### 1. App developer
+
+Someone developed a dApp and wants to support its users, since they are directly interested in the growth of the application. It could be a company, or a DAO.
+E.g.: A game developer would not want to reply to requests made by a DEX user.
+
+For this to be feasible, the enclave must be able to restrict requests to a whitelist of apps. Which is a technical feature that has to be implemented. 
+Also the wallet extension (tooling) needs to know which node to use for which call.
+
+
+#### 2. Fees
+
+Node operators could charge fees for this service.
+
+Given that everyone is now expecting this to be a free service, this is unlikely to be something that has a chance.
+
+
+#### 3. Incentives payed by the protocol.
+
+The network (or protocol) charges fees from user when submitting transactions. This is something that users expect to pay.
+
```

Comment:
user expect to pay... and why there has to be a token.

Reply:
yeah. they expect to pay for transactions to be included in the chain